src/compiler/checker.ts

```diff
@@ -47581,7 +47581,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         const target = getExportSymbolOfValueSymbolIfExported(resolveAlias(symbol));
         if (target === unknownSymbol) {
-            return true;
+            return !excludeTypeOnlyValues || !getTypeOnlyAliasDeclaration(symbol);

```

Comment:
Does this need a test for the other condition? I think the test covers the latter only.

Reply:
`!excludeTypeOnlyValues` is never covered under any test because I’m pretty sure it’s never correct to pass `false` for that. I fixed one case that forgot to pass `true` for it in https://github.com/microsoft/TypeScript/pull/55097, and the last remaining caller didn’t change any baselines when I had it pass `true`. I made a note to look at it and probably remove the parameter altogether in a follow-up PR.

Reply:
Oh yeah, the last place that doesn’t pass `true` is checking whether to elide an ImportEqualsDeclaration, which doesn’t strictly matter because it’s not a runtime error to `require()` a module that doesn’t export anything. But it should probably be elided anyway, so it’s likely that parameter will go away and we’ll treat it as if it’s always `true`.

Reply:
Sounds good, thanks!